### ISSUE TITLE
Use ~/.ssh/config or allow user to override key

### ISSUE THREAD

**Issue by aelsabbahy** [pressly/sup#86]
Awesome project!

Currently it seems ssh key is hardcoded to `~/.ssh/id_rsa` or `~/.ssh/id_dsa`.

* The easy fix: Allow user to specify it on the command line `sup -i ~/.ssh/some_key ..`
* The more complex fix: Honer settings in ~/.ssh/config

**Comment by tpaz** [comment]
Or use ssh-agent, so if you have configured your keys to be cached by ssh-add, theoretically it should work... at least in code there was preparation for that:
        // If there's a running SSH Agent, try to use its Private keys.
	sock, err := net.Dial("unix", os.Getenv("SSH_AUTH_SOCK"))
	if err == nil {
		agent := agent.NewClient(sock)
		signers, _ = agent.Signers()
	}

Not sure it works, didn't for me. I had the same problem, eventually I simply dedicated a different User to use sup to work-around it :)

**Comment by aelsabbahy** [comment]
I think ssh-agent worked for me when I tried it.

Honoring `.ssh/config` allows for more complex things like setting the user.

**Comment by VojtechVitek** [comment]
ssh-agent works, as long as `$SSH_AUTH_SOCK` env var is set correctly

otherwise, sup defaults to `~/.ssh/id_rsa` and `~/.ssh/id_dsa` --- we might want to parse more files from `~/.ssh` directory (ssh does this by default)

**Comment by VojtechVitek** [comment]
Parsing `~/.ssh/config` is also doable - but personally, I don't have time for it. Feel free to take over and send us a PR :)

**Comment by tpaz** [comment]
+1 worked for me on eventually :) to add config file support - I will first have to gain more experience in Go :)

**Comment by guitmz** [comment]
is this going to happen? :)

**Comment by jakewarren** [comment]
I opened a Bountysource bounty if anyone wants to take it on. :)

[![Bounty](https://api.bountysource.com/badge/issue?issue_id=36243899 )](https://www.bountysource.com/issues/36243899-use-ssh-config-or-allow-user-to-override-key)

**Comment by pxue** [comment]
@jakewarren I have a starter solution to parse / honour `~/.ssh/config`. shouldn't be too much of an extension to handle different keys as well

**Comment by VojtechVitek** [comment]
@pxue pls let me know if you want to add it to #105. I'll try my best to finally finish the review after it's done. (Sorry for the delay!)

**Comment by juergenhoetzel** [comment]
I suspect the *global* auth initialization code:
```go
var initAuthMethodOnce sync.Once
var authMethod ssh.AuthMethod
```

must be changed to something like:
```go
var authMethod map[string]ssh.AuthMethod
```
Do allow per-network configuration of keys?

I would prefer to specify the keys in the *supfile* instead of parsing the *openssh* configfile because I prefer to have the config in one place. I also think it is not feasible to map the openssh *CertificateFile* to sups *networks* in a consistent way.

I my pull request https://github.com/juergenhoetzel/sup/commit/74b4a4b193c0f8cc3db1bb5b85fc981bdedfbb41 I have relaxed the search for SSH-Keys: Use of *id_*" wildcard.

**Comment by VojtechVitek** [comment]
FYI, #123 has been just merged. Sup now loads any private key matching `~/.ssh/id_*` wildcard.

**Comment by VojtechVitek** [comment]
#105 and #124 were merged too.

The only remaining part is actually using the custom IdentityFile. It doesn't work yet.

**Comment by untoreh** [comment]
I think comments (`#`) in the ssh config file make sup hang

**Comment by untoreh** [comment]
Also when using a config file a `run` command is actually run only on one host (?) this is weird

**Comment by mj-ano** [comment]
Yeah, I just figured it out,  Is this a bug, not looping through the hosts if it's pulling from SSH Config file

**Comment by muodov** [comment]
For those who want to use it with custom keys, simple symlink worked for me:

```bash
$ cd ~/.ssh
$ ln -s mykey id_mykey
$ ln -s mykey.pub id_mykey.pub
```
